dropwizard-api/src/main/java/org/coner/api/entity/CompetitionGroup.java

```diff
@@ -0,0 +1,98 @@
+package org.coner.api.entity;
+
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import org.hibernate.validator.constraints.NotBlank;
+
+import javax.validation.constraints.DecimalMax;
+import javax.validation.constraints.DecimalMin;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Null;
+import java.math.BigDecimal;
+
+/**
+ * REST API entity representing a Competition Group
+ */
+@JsonPropertyOrder({"id", "name", "handicapFactor", "grouping", "resultTimeType"})
+public class CompetitionGroup extends ApiEntity {
+
+    @Null(message = "competitionGroup.id may only be assigned by the system")
+    private String id;
+    @NotBlank
+    private String name;
+    @NotNull
+    @DecimalMin("0.000")
+    @DecimalMax("1.000")
+    private BigDecimal handicapFactor;
+    @NotNull
+    private Boolean grouping;
+    @NotBlank
+    private String resultTimeType;
```

Comment:
I agree validation for this would be very helpful. I knew something was wrong when I found myself referring to the Domain entity source while writing the API test fixtures...